40-0247 - Write automated tests

tlei123 opened the pull request:
\[Making this Ready for Review in case we decide to merge as is, and open follow-up tickets for the outstanding file-upload page\]

## Summary

- Write automated tests for Pres. Mem. Cert. (40-0247)
- Team: Veteran Facing Forms
- No Flipper yet; form's still in-development

## Related issue(s)

- _Link to ticket created in va.gov-team-forms repo_
department-of-veterans-affairs/va.gov-team-forms#481

## Testing done

- All tests passed locally

## What areas of the site does it impact?

This form ONLY

## Acceptance criteria

## Requested Feedback

Beyond the basic code-review, I also need guidance on how to fix the coverage-report.  Even though all unit-tests were in-place for all pages and all passed when I ran `yarn test:unit src/applications/simple-forms/40-0247/tests/unit/**/*.unit.spec.jsx`, the Pages coverage sub-report's showing some confusing results, as if the v3-web-components pages' unit-tests, and the non-v3 file-upload page's, hadn't run at all.  \[See screenshot below for the Pages sub-report\]

![image](https://github.com/department-of-veterans-affairs/vets-website/assets/587583/04b8a922-48cd-49c0-92ca-1dced57efb2a)

tlei123 committed:
Add test data for req'd Vet DoB and DoD

tlei123 committed:
Revert 40-0247 Add'l Certs Req page to v1 components
Page has v1 number-input, so can't be mixed with v3 address

tlei123 committed:
Support title functions in address-definition label param
Some address-page uiSchemas require JSX for the label

tlei123 committed:
Fix 40-0247 Add'l Certs Req addr-label custom-styling

tlei123 committed:
Add 40-0247 page unit-tests

tlei123 committed:
Merge branch 'main' into va-forms/40-0247/481-write-tests

tlei123 committed:
Assign 40-0247 applicantFullName to ConfirmationPageView submitierName prop

tlei123 committed:
Add 40-2470 submit-transformer and containers unit-tests

tlei123 committed:
WIP Update 40-0247 e2e-spec

tlei123 committed:
Merge branch 'main' into va-forms/40-0247/481-write-tests

tlei123 committed:
Finish updating 40-0247 e2e-spec

tlei123 committed:
Merge branch 'main' into va-forms/40-0247/481-write-tests

tlei123 committed:
Remove unused helpers-file from 40-0247 app

tlei123 committed:
Fix 40-0247 unit-tests

tlei123 committed:
Add 40-0247 Vet Support Docs page unit-test

tlei123 commented on src/applications/simple-forms/shared/tests/e2e/helpers.js:
```diff
@@ -1,3 +1,19 @@
+export const getPagePaths = formConfig => {
```
This'll save us from manually writing out page-paths for v3-web-component unit-tests.

rhasselle-oddball reviewed (commented):
changes overall look good. yea as for the code coverage thing -
code coverage only looks for things like functions, lambdas, statements with conditionals. so it doesn't cover simple constants of schemas. So its entirely possible to have entire files with no branching logic, and thus there's nothing for code coverage to check.

I wonder if it was defined as a const, and then exported, vs just directly exported, then it might count toward code coverage?

tlei123 commented on src/applications/simple-forms/40-0247/pages/additionalCertificatesRequest.js:
```diff
@@ -1,25 +1,23 @@
-import {
-  addressNoMilitaryUI,
-  addressNoMilitarySchema,
-} from 'platform/forms-system/src/js/web-component-patterns/addressPattern.jsx';
```
It's 'cuz we have a non-v3 qty-field below the address stuff, so i can't mix the v3 address in there.

tlei123 replied:
ok, i'm gonna change the whole page to v3 now that i can copy-n-paste you v3 number-patterns.  i'll keep you posted also on my coverage-weirdness.

tlei123 commented:
> I wonder if it was defined as a const, and then exported, vs just directly exported, then it might count toward code coverage?

@rhasselle-oddball , I just double-checked my page-files and they're all exported directly, no const.

tlei123 committed:
Merge branch 'main' into va-forms/40-0247/481-write-tests

tlei123 committed:
Convert 40-0247 Add'l Certs Req page to v3-web-components

tlei123 committed:
Fix unit-test filename

tlei123 committed:
Refactor 40-2470 Add'l Certs Req page unit-test

tlei123 committed:
Customize 40-0247 address pages street2 labels

tlei123 committed:
Modularize 40-0247 Add'l Certs Req page custom-validation
Moving code out of the 'ui:validations' prop makes unit-testing the validation easier

tlei123 committed:
Add 40-0247 validations unit-test

tlei123 committed:
Modularize 40-0247 Certs page reviewField
Taking the function out of page-config makes it easier to unit-test

tlei123 committed:
Merge branch 'main' into va-forms/40-0247/481-write-tests

tlei123 committed:
Update 40-0247 test-data

tlei123 committed:
Add/refactor 40-0247 form/page configs and unit-tests

tlei123 commented:
@rhasselle-oddball , the _aggregate_ coverage-percentages are looking good \[see screenshot below\].  The file-upload page is still WIP right now.  I only have our 2 regular fields/errors tests in-place for this page, but I already have a checklist-item for it in my follow-up ticket [#585](https://github.com/department-of-veterans-affairs/VA.gov-team-forms/issues/585).

Let me know if the aggregate numbers are sufficient to for approval.  Thanks much!

![image](https://github.com/department-of-veterans-affairs/vets-website/assets/587583/f94138d3-9853-4bdc-b649-73bc1c927c8e)

tlei123 committed:
Mock 40-0427 form submit

tlei123 committed:
Merge branch 'main' into va-forms/40-0247/481-write-tests

tlei123 committed:
Merge branch 'main' into va-forms/40-0247/481-write-tests

tlei123 committed:
Merge branch 'main' into va-forms/40-0247/481-write-tests

tlei123 committed:
Fix typo

tlei123 committed:
Undo last commit

rhasselle-oddball reviewed (approved):
lots of tests! nice

tlei123 committed:
Merge branch 'main' into va-forms/40-0247/481-write-tests

tlei123 committed:
Merge branch 'main' into va-forms/40-0247/481-write-tests

tlei123 committed:
Merge branch 'main' into va-forms/40-0247/481-write-tests

tlei123 committed:
Merge branch 'main' into va-forms/40-0247/481-write-tests

tlei123 committed:
Merge branch 'main' into va-forms/40-0247/481-write-tests

tlei123 committed:
Merge branch 'main' into va-forms/40-0247/481-write-tests

tlei123 committed:
Merge branch 'main' into va-forms/40-0247/481-write-tests

tlei123 committed:
Merge branch 'main' into va-forms/40-0247/481-write-tests

tlei123 committed:
Merge branch 'main' into va-forms/40-0247/481-write-tests

tlei123 committed:
Merge branch 'main' into va-forms/40-0247/481-write-tests
